### ISSUE TITLE
NoSuchMethodError

### ISSUE THREAD

**Issue by Spurlos** [klesun/deep-js-completion#20]
Plugin version: 2020.11.30.001
PhpStorm 2021.1.1
Build #PS-211.7036.8, built on April 15, 2021
Runtime version: 11.0.10+9-b1341.35 amd64

```
java.lang.NoSuchMethodError: 'com.intellij.lang.javascript.psi.JSType com.intellij.lang.javascript.psi.JSTypeEvaluationResult.getType()'
	at org.klesun.deep_js_completion.contexts.SearchCtx.$anonfun$getWsType$5(SearchCtx.scala:59)
	at scala.Option.flatMap(Option.scala:283)
	at org.klesun.deep_js_completion.contexts.SearchCtx.getWsType(SearchCtx.scala:59)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:198)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:66)
	at org.klesun.deep_js_completion.completion_providers.PropNamePvdr.$anonfun$getGotoDeclarationTargets$4(PropNamePvdr.scala:342)
	at scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:585)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:599)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:592)
	at scala.collection.Iterator$$anon$6.hasNext(Iterator.scala:470)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:591)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:592)
	at scala.collection.Iterator$$anon$6.hasNext(Iterator.scala:470)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:335)
	at org.klesun.deep_js_completion.completion_providers.PropNamePvdr.getGotoDeclarationTargets(PropNamePvdr.scala:349)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt.fromGTDProvidersInner(gtdProviders.kt:31)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt.access$fromGTDProvidersInner(gtdProviders.kt:1)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt$fromGTDProviders$1.invoke(gtdProviders.kt:18)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt$fromGTDProviders$1.invoke(gtdProviders.kt)
	at com.intellij.codeInsight.navigation.impl.CommonKt.processInjectionThenHost(common.kt:31)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt.fromGTDProviders(gtdProviders.kt:17)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationOrUsageHandler2.gotoDeclarationOrUsages(GotoDeclarationOrUsageHandler2.kt:29)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationOrUsageHandler2.getCtrlMouseInfo(GotoDeclarationOrUsageHandler2.kt:35)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationAction.getCtrlMouseInfo(GotoDeclarationAction.java:93)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.doExecute(CtrlMouseHandler.java:392)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.lambda$execute$0(CtrlMouseHandler.java:370)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:521)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$attemptComputation$3(NonBlockingReadActionImpl.java:468)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1096)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:79)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:157)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:119)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:57)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:57)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:116)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:79)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.attemptComputation(NonBlockingReadActionImpl.java:486)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$1(NonBlockingReadActionImpl.java:408)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

```
java.lang.NoSuchMethodError: 'com.intellij.lang.javascript.psi.JSType com.intellij.lang.javascript.psi.JSTypeEvaluationResult.getType()'
	at org.klesun.deep_js_completion.contexts.SearchCtx.$anonfun$getWsType$5(SearchCtx.scala:59)
	at scala.Option.flatMap(Option.scala:283)
	at org.klesun.deep_js_completion.contexts.SearchCtx.getWsType(SearchCtx.scala:59)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:198)
	at org.klesun.deep_js_completion.contexts.ExprCtx.$anonfun$findExprType$1(ExprCtx.scala:54)
	at org.klesun.deep_js_completion.contexts.ExprCtx.$anonfun$findExprType$1$adapted(ExprCtx.scala:53)
	at scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:585)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:599)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:591)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at org.klesun.lang.DeepJsLang$MemIt$$anon$1.hasNext(DeepJsLang.scala:223)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:592)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:591)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2(DeepJsLang.scala:412)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2$adapted(DeepJsLang.scala:412)
	at scala.collection.Iterator$$anon$6.hasNext(Iterator.scala:472)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:335)
	at org.klesun.lang.DeepJsLang$.frs(DeepJsLang.scala:413)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:199)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:66)
	at org.klesun.deep_js_completion.completion_providers.PropNamePvdr.$anonfun$getGotoDeclarationTargets$4(PropNamePvdr.scala:342)
	at scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:585)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:599)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:592)
	at scala.collection.Iterator$$anon$6.hasNext(Iterator.scala:470)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:591)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:592)
	at scala.collection.Iterator$$anon$6.hasNext(Iterator.scala:470)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:335)
	at org.klesun.deep_js_completion.completion_providers.PropNamePvdr.getGotoDeclarationTargets(PropNamePvdr.scala:349)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt.fromGTDProvidersInner(gtdProviders.kt:31)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt.access$fromGTDProvidersInner(gtdProviders.kt:1)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt$fromGTDProviders$1.invoke(gtdProviders.kt:18)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt$fromGTDProviders$1.invoke(gtdProviders.kt)
	at com.intellij.codeInsight.navigation.impl.CommonKt.processInjectionThenHost(common.kt:31)
	at com.intellij.codeInsight.navigation.impl.GtdProvidersKt.fromGTDProviders(gtdProviders.kt:17)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationOrUsageHandler2.gotoDeclarationOrUsages(GotoDeclarationOrUsageHandler2.kt:29)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationOrUsageHandler2.getCtrlMouseInfo(GotoDeclarationOrUsageHandler2.kt:35)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationAction.getCtrlMouseInfo(GotoDeclarationAction.java:93)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.doExecute(CtrlMouseHandler.java:392)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.lambda$execute$0(CtrlMouseHandler.java:370)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:521)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$attemptComputation$3(NonBlockingReadActionImpl.java:468)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1096)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:79)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:157)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:119)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:57)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:57)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:116)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:79)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.attemptComputation(NonBlockingReadActionImpl.java:486)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$1(NonBlockingReadActionImpl.java:408)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by klesun** [comment]
Related: https://github.com/klesun/deep-js-completion/issues/19

**Comment by klesun** [comment]
Released fix in [2021.04.21.001](https://plugins.jetbrains.com/plugin/11478-deep-js-completion/versions/stable/117651)

It should become public in about two days - when JetBrains is done validating
